Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -305,14 +411,15 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            for header in self.headers {
-                if self.buffer.equalCaseInsensitiveASCII(view: "connection".utf8, at: header.name) {
-                    if self.buffer.equalCaseInsensitiveASCII(view: "close".utf8, at: header.value) {
-                        return false
-                    }
-                    return self.buffer.equalCaseInsensitiveASCII(view: "keep-alive".utf8, at: header.value)
-                }
-            }
+            var keepAlive: Bool = false
+            var close: Bool = false
+            (keepAlive, close) = self.parseKeepAliveAndClose()
+            
+            // TODO: Handle the case where both keep-alive and close are used
+            
+            if keepAlive { return true }
+            if close { return false }
```

Comment:
Let's push this logic down into `parseKeepAliveAndClose`, such that it returns only a `Bool`.